### PR TITLE
fix: fixed adding/removing address funcionality

### PR DESCRIPTION
**Pull request by Frodigo**
## Description
- fixed adding address on my account
- fixed removing address on my account

## Related Issue
#765 #766

## Motivation and Context
bugfixing

## How Has This Been Tested?
tested on my account:
1. log-in as a customer
2. go to address details
3. add a new address
4. delete address

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/components/MyAccount/AddressForm.vue [hunk 1]
```diff
@@ -273,6 +273,10 @@ export default defineComponent({
     const countries = ref([]);
     const country = ref(null);
 
+    const reloadCountry = async (params) => {
```
Comment (bartoszherba) [inline packages/theme/components/MyAccount/AddressForm.vue:276]: I am not sure if this should block further script execution.

**Review by bartoszherba — APPROVED** [review]
One comment, nothing critical I think.

File: packages/theme/components/MyAccount/AddressForm.vue (hunk 1 above)
Comment (sethidden) [reply packages/theme/components/MyAccount/AddressForm.vue:276]: This is a function that awaits the promise only if it's called, it'll run only on user input after the initial hydration finished

File: packages/theme/pages/MyAccount/AddressesDetails.vue [hunk 2]
```diff
@@ -120,6 +120,16 @@ export default defineComponent({
     const getTranslatedUrlAddress = (title) => app.i18n.t(`${title}`).toLowerCase().replace(' ', '-');
     const isNewAddress = computed(() => route.value.query.id === 'new');
     const editingAddress = computed(() => !!route.value.query.id);
+    const { fetch } = useFetch(async () => {
+      const addressesData = await load();
+      userAddresses.value = userAddressesGetters.getAddresses(addressesData);
+
+      const activeAddressData = userAddresses.value
+        .filter((address) => String(address?.id) === route.value.query.id)
+        .pop();
```
Comment (sethidden) [inline packages/theme/pages/MyAccount/AddressesDetails.vue:129]:
```suggestion
      const [ ..._front, activeAddressData] = userAddresses.value
        .filter((address) => String(address?.id) === route.value.query.id);
```

```suggestion
      const [ ...front, activeAddressData] = userAddresses.value
        .filter((address) => String(address?.id) === route.value.query.id).at(-1);
```

.pop() seems ununtuitive to me here, proposed some more explicit ways

File: packages/theme/pages/MyAccount/AddressesDetails.vue [hunk 3]
```diff
@@ -135,7 +145,7 @@ export default defineComponent({
       const isDefault = userAddressesGetters.isDefault(address);
       if (!isDefault) {
         await remove({ address });
-        await load({});
+        fetch();
```
Comment (sethidden) [inline packages/theme/pages/MyAccount/AddressesDetails.vue:148]:
```suggestion
        await fetch();
```

Is dropping the await intentional here? Could be problematic if someone added something new behind the fetch later

**Review by sethidden — CHANGES_REQUESTED** [review]

File: packages/theme/pages/MyAccount/AddressesDetails.vue (hunk 2 above)
Comment (Frodigo) [reply packages/theme/pages/MyAccount/AddressesDetails.vue:129]: thanks

File: packages/theme/pages/MyAccount/AddressesDetails.vue (hunk 3 above)
Comment (Frodigo) [reply packages/theme/pages/MyAccount/AddressesDetails.vue:148]: fetch retriggers useFetch and it does not return a promise so it cannot be awaited.

File: packages/theme/pages/MyAccount/AddressesDetails.vue (hunk 2 above)
Comment (Frodigo) [reply packages/theme/pages/MyAccount/AddressesDetails.vue:129]: but after change you requested code stopped work as excepted

**Review by sethidden — APPROVED** [review]

**Pull request merged**